### PR TITLE
[ADD] new module sale_order_lot_selection_by_quant

### PR DESCRIPTION
**Pull request by HekkiMelody**
New module letting the user select a lot by choosing a specific quant through a widget.

### CODE REVIEW

**Comment by francesco-ooops** [comment]
@grindtildeath slightly related to https://github.com/OCA/stock-logistics-workflow/pull/1391 , what do you think of this one? :)

**Review by geomer198 — APPROVED** [review]
LGTM

**Review by francesco-ooops — APPROVED** [review]
Functional ok!

**Comment by francesco-ooops** [comment]
@simahawk do you like this one? :)

**Comment by francesco-ooops** [comment]
@pedrobaeza can this be merged?

File: sale_order_lot_selection_by_quant/readme/ROADMAP.rst
Comment (pedrobaeza) [inline sale_order_lot_selection_by_quant/readme/ROADMAP.rst:1]: Don't add empty files.

**Review by pedrobaeza — COMMENTED** [review]
Why do you need JS?

You may want to add some screenshots for the README

**Comment by HekkiMelody** [comment]
> Why do you need JS?

My first attempted involved only adding a `quant_id` field and then changing the `lot_id` based on the selected quant. However, I think the user experience was too poor for it to be usable.

- The name of the quant was not good enough on its own. I would have had to override `name_get` to add some info such as warehouse.
- For the use case I'm covering, the user always wants to click "search more" to see info such as availability of the product in various warehouses
- Sales Order Lines are quite busy enough as it is, so I figured this solution would also have been cleaner. 
- With not much refactoring it's easily reusable or expandable into a more generic widget.

There were other problems too. Going the JS route was not my first solution either.

**Comment by pedrobaeza** [comment]
OK, thanks. Merging based on reviews and comments:

/ocabot merge nobump

**Pull request merged**